script/testing/reporting/parsers/parse_data.py

```diff
@@ -106,6 +107,59 @@ def _parse_jenkins_env_vars():
     }
     return metadata
 
+def _get_git_commit_id(git_branch):
+    """
+    Get the commit hash.
+    
+    Parameters
+    ----------
+    git_branch : str
+        The branch that the code is from. For pull-requests this will be 
+        PR-####.
+
+    Returns
+    -------
+    str
+        The commit hash for the latest commit on the branch.
+    """
+    pr_number_match = re.search('PR-\d+', git_branch)
+    if pr_number_match:
+        # Get the hash from the last commit to the PR.
+        try:
+            pr_number = int(pr_number_match.group(1))
+            commits = _get_git_pr_commits(pr_number)
+            latest_commit = commits[-1]
```

Comment:
Read the PR text for why this might not be the best solution and my proposed alternative.